components/segmenter_lstm/src/lstm.rs

```diff
@@ -0,0 +1,157 @@
+use crate::error::Error;
+use crate::math_helper;
+use crate::structs;
+use ndarray::{Array1, Array2, ArrayBase, Dim, ViewRepr};
+use std::str;
+use unicode_segmentation::UnicodeSegmentation;
+
+pub struct Lstm {
+    data: structs::LstmData,
+}
+
+#[derive(Debug)]
+pub struct TestText {
+    pub data: structs::TestTextData,
+}
+
+impl TestText {
+    pub fn new(data: structs::TestTextData) -> Self {
+        Self { data }
+    }
+}
+
+impl Lstm {
+    /// `try_new` is the initiator of struct `Lstm`
+    pub fn try_new(data: structs::LstmData) -> Result<Self, Error> {
+        if data.dic.len() > std::i16::MAX as usize {
+            return Err(Error::Limit);
+        }
+        Ok(Self { data })
```

Comment:
Add one more check for model validity:

```
if !self.data.model.contains("codepoints") && !self.data.model.contains("graphclust") {
    return Err(Error::Syntax);
}
```

Reply:
I did the following, to make sure names such as `Thai_codepointssss_exclusive_model4_heavy` don't pass.

```
if !data.model.contains("_codepoints_") && !data.model.contains("_graphclust_") {
            return Err(Error::Syntax);
        }
```